src/main/java/redis/clients/jedis/BinaryJedisCluster.java

```diff
@@ -28,11 +29,22 @@
     MultiKeyBinaryJedisClusterCommands, JedisClusterBinaryScriptingCommands, Closeable {
 
   public static final int HASHSLOTS = 16384;
-  protected static final int DEFAULT_TIMEOUT = 2000;
-  protected static final int DEFAULT_MAX_ATTEMPTS = 5;
+
+  /**
+   * Default timeout in milliseconds.
+   */
+  public static final int DEFAULT_TIMEOUT = 2000;
+  public static final int DEFAULT_MAX_ATTEMPTS = 5;
 
   protected int maxAttempts;
 
+  /**
+   * After this amount of time we will do no more retries and report the operation as failed.
+   *
+   * Defaults to {@link #DEFAULT_TIMEOUT} if unset, or {@code soTimeout} if available.
```

Comment:
Where does it default to DEFAULT_TIMEOUT?

Reply:
It's pretty opaque, but If you follow the `DEFAULT_TIMEOUT`'s way from the `public BinaryJedisCluster(Set<HostAndPort> nodes)` constructor, you can see that it eventually makes its way into the `maxTotalRetriesDuration` instance variable.